### PR TITLE
Deployment previews on pull requests

### PR DESCRIPTION
**Pull request by jasonbarry**
I think this repo would benefit in having deployment previews on every pull request, that way we can generate an automatic build of the website per PR rather than having to checkout branches locally to review/verify changes.

These are the changes needed to integrate w/ [FeaturePeek](https://featurepeek.com), a deployment preview provider. If the GitHub App is installed (for webhooks), then every subsequent PR will have its own deployment preview.

### CODE REVIEW

File: .circleci/config.yml
Comment (internettrans) [inline .circleci/config.yml]: Would probably be best to only run this on non-master branches, since the master branch is going to be deployed to the live site.

**Review by internettrans — COMMENTED** [review]
I like it, thanks. Could you share an example of a preview site on peek.run? How will we know the link to the peek.run preview for each pull request?

**Comment by jasonbarry** [comment]
@joeldenning here's an example: https://peek.run/902hn27j 

When the [GitHub App](https://github.com/apps/featurepeek) is installed, the FeaturePeek bot will post a comment that contains the link.

**Comment by internettrans** [comment]
I installed the Github app for this repository. Once the previews are only deployed for non-master branches, I'll approve and merge.

File: .circleci/config.yml
Comment (jasonbarry) [reply .circleci/config.yml]: @joeldenning I've added a line to ignore the master branch on the `build-website` job – please double-check that that's correct.

File: .circleci/config.yml
Comment (internettrans) [inline .circleci/config.yml]: I think this will break things, since we need the build to run on the master branch. It might make sense to make a new job called `deploy-preview` and then do the branch filter on that job instead of on the `build-website` job

Comment (jasonbarry) [reply .circleci/config.yml]: @joeldenning take a look at the latest diff – I put a condition inside the existing `build-website` job (inspired by https://circleci.com/docs/2.0/configuration-reference/#ending-a-job-from-within-a-step)

**Review by internettrans — APPROVED** [review]
Looks good, thanks. I think that circle ci is only configured to run on branches on the main repository, not forks. So we might need to fix that, too. I am going to test the peek preview with a branch right after merging this pull request

**Comment by jasonbarry** [comment]
@joeldenning cool! If you create a free account on https://dashboard.featurepeek.com, you can claim the `single-spa` team since you're in the org. Let me know how it goes!

**Comment by internettrans** [comment]
I went to create an account with feature peek, but it asked for more permissions than I am comfortable with.

image

Is it possible to use FeaturePeek without giving full write access to the repository and its settings?

**Pull request merged**
